rasa/core/test.py

```diff
@@ -352,7 +352,11 @@ async def _create_data_generator(
     test_data_importer = TrainingDataImporter.load_from_dict(
         training_data_paths=[resource_name]
     )
-    story_graph = await test_data_importer.get_stories(use_e2e=use_e2e)
+    if use_e2e:
```

Comment:
I think this branching is confusing, as we are also passing in a variable `resource_name` into the importer. It's hard to know how those two different things interplay. Would it not be clearer if we do this resource selection up higher when we start the test process, and then `use_e2e` could remain solely for determining how to read the file?

Reply:
Very good point. The `TrainingDataImporter` is already categorizing files (e.g. into `nlu` and `stories`) and hence I think this should be part of the `TrainingDataImporter`. As this function `_create_data_generator` is called from different entrypoints I found it preferable to handle this in one place instead of multiple ones. Otherwise we might run into situations as we had it before where e.g. tests or also the server weren't respecting the conditions we have on these test files (e.g. the `test_` prefix). 

I agree that this is all confusing though 🙈 This still seemed the somewhat clearest approach to me though.

Reply:
Maybe we should use a different name than `use_e2e` then? As it normally dictates what reader to use.
e.g. `conversation_test_stories`?

Reply:
You mean renaming it within `_create_data_generator` or in general in the entire module? I'd be open to rename it for `_create_data_generator` but unfortunately `use_e2e` has multiple other implications on the testing

Reply:
> but unfortunately use_e2e has multiple other implications on the testing

This is why it might make sense to rename. You've added it to lots of methods in this PR with the sole aim of selecting different files in this method. Whereas it normally has the meaning of how a markdown file is read. When we remove markdown, and hopefully a lot of usages of `use_e2e`, we will still need this one to select conversation tests.